Source/Details/ASRangeController.mm

```diff
@@ -253,30 +223,26 @@ - (void)_updateVisibleNodeIndexPaths
   if ((!_preserveCurrentRangeMode && ASInterfaceStateIncludesVisible(selfInterfaceState)) || [[self class] isFirstRangeUpdateForRangeMode:rangeMode]) {
     rangeMode = [ASRangeController rangeModeForInterfaceState:selfInterfaceState currentRangeMode:_currentRangeMode];
   }
-
-  ASRangeTuningParameters parametersPreload = [_layoutController tuningParametersForRangeMode:rangeMode
-                                                                                      rangeType:ASLayoutRangeTypePreload];
-  if (ASRangeTuningParametersEqualToRangeTuningParameters(parametersPreload, ASRangeTuningParametersZero)) {
-    preloadIndexPaths = visibleIndexPaths;
-  } else {
-    preloadIndexPaths = [_layoutController indexPathsForScrolling:scrollDirection
-                                                          rangeMode:rangeMode
-                                                          rangeType:ASLayoutRangeTypePreload];
-  }
   
-  ASRangeTuningParameters parametersDisplay = [_layoutController tuningParametersForRangeMode:rangeMode
-                                                                                    rangeType:ASLayoutRangeTypeDisplay];
-  if (rangeMode == ASLayoutRangeModeLowMemory) {
-    displayIndexPaths = [NSSet set];
-  } else if (ASRangeTuningParametersEqualToRangeTuningParameters(parametersDisplay, ASRangeTuningParametersZero)) {
-    displayIndexPaths = visibleIndexPaths;
```

Comment:
This deletes some fast paths that are very valuable performance optimizations, by avoiding extra set assembly & object manipulations when some sets are truly identical.  In the most main-thread-performance sensitive views, it's always made a big difference to set the tuning parameters to the same value for display and preload.

Although the core of this change does help bring down the time for generating two unique sets, I re-measured the benefit of this set-sharing after my most recent change and so included the more descriptive BOOL cases to handle this.

Because ASRangeController is, essentially, the core of AsyncDisplayKit — it's crucial that we aggressively optimize this path.  Most importantly, this path currently occurs during the same runloop as other very large operations like [UICollectionView layoutSubviews], so each ms tends to contribute directly to additional frame drops.  Combined with the high frequency of the method being called, it is still responsible for >10% of all main thread time (and a greater proportion of "time over 16.7ms")